src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -331,6 +340,9 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
                     disable_cluster_connect(cmd, client, resource_group_name, cluster_name, kube_config, kube_context, values_file, release_namespace, helm_client_location)
                 return cc_response
             else:
+                # If cluster is of kind provisioned cluster, there are several properties that cannot be updated
+                validate_existing_provisioned_cluster_for_reput(preview_cluster_resource, kubernetes_distro, kubernetes_infra, enable_private_link, private_link_scope_resource_id, distribution_version, azure_hybrid_benefit, location)    
+
```

Comment:
Actually, why is this validation needed, since this seems to be the error case.  Moreover, you can probably do this error check first in an if block and move the current if block out, below the new if block - as the default.

Reply:
you are right, that validation was not put there on purpose - there is already an appropriate error message thrown